### PR TITLE
fix: DbConnection cannot be extended

### PR DESCRIPTION
**Pull request by cristianIOHK**
Signed-off-by: Cristian G <cristian.castro@iohk.io>

### Description: 
This PR provides the DbConnection as an interface for the end to end tests to use a custom DbConnection to use in memory db.

### Checklist: 
- [x] My PR follows the [contribution guidelines](https://github.com/input-output-hk/atala-prism-wallet-sdk-kmm/blob/main/CONTRIBUTING.md) of this project
- [x] My PR is free of third-party dependencies that don't comply with the [Allowlist](https://toc.hyperledger.org/governing-documents/allowed-third-party-license-policy.html#approved-licenses-for-allowlist)
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked the PR title to follow the [conventional commit specification](https://www.conventionalcommits.org/en/v1.0.0/)

### CODE REVIEW

File: edge-agent-sdk/src/commonMain/kotlin/org/hyperledger/identus/walletsdk/pluto/PlutoImpl.kt [hunk 1]
```diff
@@ -53,7 +53,7 @@ import org.hyperledger.identus.walletsdk.pluto.data.StorableCredential as Storab
  * @property db The instance of `SdkPlutoDb` representing the connection to the database.
  * @property isConnected A flag to indicate whether the database connection is established or not.
  */
-class PlutoImpl(private val connection: DbConnection) : Pluto {
+class PlutoImpl(private val connection: DbConnectionImpl) : Pluto {
```
Comment (amagyar-iohk) [inline edge-agent-sdk/src/commonMain/kotlin/org/hyperledger/identus/walletsdk/pluto/PlutoImpl.kt:56]: This should has the interface type

**Review by amagyar-iohk — CHANGES_REQUESTED** [review]

File: edge-agent-sdk/src/commonMain/kotlin/org/hyperledger/identus/walletsdk/pluto/PlutoImpl.kt (hunk 1 above)
Comment (amagyar-iohk) [inline edge-agent-sdk/src/commonMain/kotlin/org/hyperledger/identus/walletsdk/pluto/PlutoImpl.kt:56]:
```suggestion
class PlutoImpl(private val connection: DbConnection) : Pluto {
```

File: edge-agent-sdk/src/commonMain/kotlin/org/hyperledger/identus/walletsdk/pluto/PlutoImpl.kt [hunk 2]
```diff
@@ -34,7 +34,7 @@ import org.hyperledger.identus.walletsdk.domain.models.keyManagement.JWK
 import org.hyperledger.identus.walletsdk.domain.models.keyManagement.PrivateKey
 import org.hyperledger.identus.walletsdk.domain.models.keyManagement.StorableKey
 import org.hyperledger.identus.walletsdk.pluto.backup.models.BackupV0_0_1
-import org.hyperledger.identus.walletsdk.pluto.data.DbConnection
+import org.hyperledger.identus.walletsdk.pluto.data.DbConnectionImpl
```
Comment (amagyar-iohk) [inline edge-agent-sdk/src/commonMain/kotlin/org/hyperledger/identus/walletsdk/pluto/PlutoImpl.kt:37]:
```suggestion
import org.hyperledger.identus.walletsdk.pluto.data.DbConnection
```

File: edge-agent-sdk/src/commonMain/kotlin/org/hyperledger/identus/walletsdk/pluto/PlutoImpl.kt (hunk 1 above)
Comment (cristianIOHK) [reply edge-agent-sdk/src/commonMain/kotlin/org/hyperledger/identus/walletsdk/pluto/PlutoImpl.kt]: Done

**Pull request merged**